pkg/kubelet/secret/secret_manager.go

```diff
@@ -169,7 +169,14 @@ func (s *secretStore) Get(namespace, name string) (*v1.Secret, error) {
 	data.Lock()
 	defer data.Unlock()
 	if data.err != nil || !s.clock.Now().Before(data.lastUpdateTime.Add(s.ttl)) {
-		secret, err := s.kubeClient.Core().Secrets(namespace).Get(name, metav1.GetOptions{})
+		opts := metav1.GetOptions{}
+		if data.secret != nil && data.err == nil {
+			// This is just a periodic refresh of a secret we successfully fetched previously.
+			// In this case, we just try to get data from apiserver cache to reduce the load
+			// on etcd (the cache is eventuallly consistent).
+			opts.ResourceVersion = "0"
```

Comment:
@yujuhong - sure, we should definitely merge this one and see the results of this improvment before the other PR.